client/src/components/TaskRunModal/proxy/ProxyRunDetailsMessagesView.tsx

```diff
@@ -20,10 +20,17 @@ export function ProxyRunDetailsMessagesView(props: Props) {
     return new Error(error?.message);
   }, [error]);
 
+  const headerText = useMemo(() => {
```

Comment:
should not be in a memo since the computation is really simple. We should likely add a mention in the `AGENTS.md`

```
- Do not use `useMemo` when the computation is simple and the result is a scalar (string, number) since it would unnecessarily add overhead.
```